lib/doc_handler.dart

```diff
@@ -0,0 +1,156 @@
+// Copyright (c) 2015, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+
+library dartpad.doc_handler;
+
+import 'dart:convert';
+import 'dart:html';
+import 'dart:math' as math;
+
+import 'package:markd/markdown.dart' as markdown;
+
+import 'editing/editor.dart';
+import 'context.dart';
+import 'services/common.dart';
+import 'src/ga.dart';
+import 'dart_pad.dart';
+import 'dartservices_client/v1.dart';
+
+class DocHandler {
+  static const List cursorKeys = const [
+    KeyCode.LEFT,
+    KeyCode.RIGHT,
+    KeyCode.UP,
+    KeyCode.DOWN
+  ];
+  Analytics _ga;
+  Editor _editor;
+  Context _context;
+
+  final NodeValidatorBuilder _htmlValidator = new NodeValidatorBuilder.common()
+    ..allowElement('a', attributes: ['href'])
+    ..allowElement('img', attributes: ['src']);
+
+  DocHandler(this._editor, this._context, this._ga);
+
+  void generateDoc(DivElement docPanel) {
+    if (!(_context.focusedEditor == 'dart'
+        && _editor.hasFocus
+        && _editor.document.selection.isEmpty)) {
+      return;
+    }
+    _ga.sendEvent('main', 'help');
```

Comment:
Let's move this call to `ga` to playground, when the F1 key is hit. We want to track when the user does something explicitly. Also, it seems odd to send analytics as a side effect of creating a documentation div.